File: Leetcode/findTheDifference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) 
    {
        std::sort(s.begin(), s.end());
        std::sort(t.begin(), t.end());
        bool lastchar = true;
        char op = '0';
        for(int z = 0; z < s.size(); z++)
        {
            if(s[z] != t[z])
            {
                op = t[z];
                lastchar = false;
            }
        }
        if(lastchar)
        {
            op = t[s.size()];
        }
        return op;
    }
};

File: Leetcode/Find-the-Difference.py
class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        container = {}
        for x in s:
            if x in container:
                container[x] += 1
            else:
                container[x] = 1
        for x in t:
            if x not in container:
                return x
            else:
                container[x] -= 1
                if container[x] < 0:
                    return x
                
